Prepare CHANGELOG.md for v1.0.1 release (#2433)
spiffe/spire: CHANGELOG.md

# Changelog

> ## [1.0.1] - 2021-08-02

Reviewer: We may have here the expected date of the release (if no issues are found, I think we plan to release on Thursday 2021-08-05)? We can update it if needed later.

### Added

> - A new `tpm_devid` NodeAttestor plugin is now available (#2111, #2427)

Reviewer: Although it can be inferred from the plugin name, I think that it would be good to have something in this line that briefly describes what kind of node attestation is being introduced. Probably indicating that's a plugin that attests nodes that own a TPM and that have been provisioned with a DevID certificate or something along those lines.

Reviewer 2: How about this

```suggestion
- LDevID-based TPM attestation can now be performed via a new `tpm_devid` NodeAttestor plugin (#2111, #2427)
```

- Caller details are now logged for unauthorized Server API calls (#2399)

> - The `aws_iid` NodeAttestor plugin can now be configured to assume an AWS IAM role for requests to AWS (#2387)

Reviewer 2: ```suggestion
- The `aws_iid` NodeAttestor plugin now supports attesting nodes across multiple AWS accounts via AWS IAM role assumption (#2387)
```

- Added support for running the `k8s_sat` NodeAttestor plugin with Kubernetes v1.21 (#2423)
- Call counter metrics are now emitted for SPIRE Server rate limiters (#2422)

> - SPIRE Server now logs a message on startup when configured TTL values may result in unexpectedly shorter CA or SVID TTLs (#2284)

Reviewer 2: Applies only to SVIDs (CAs won't be cut short by these controls).

```suggestion
- SPIRE Server now logs a message on startup when configured TTL values may result in SVIDs with a shorter lifetime than expected (#2284)
```
